vault/datadog_checks/vault/vault.py

```diff
@@ -0,0 +1,170 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from time import time as timestamp
+
+import requests
+
+from datadog_checks.checks import AgentCheck
+from datadog_checks.config import _is_affirmative
+from datadog_checks.utils.containers import hash_mutable
+from .errors import ApiUnreachable
+
+
+class Vault(AgentCheck):
+    CHECK_NAME = 'vault'
+    DEFAULT_API_VERSION = '1'
+    EVENT_LEADER_CHANGE = 'vault.leader_change'
+    SERVICE_CHECK_CONNECT = 'vault.can_connect'
+    SERVICE_CHECK_UNSEALED = 'vault.unsealed'
+    SERVICE_CHECK_INITIALIZED = 'vault.initialized'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(Vault, self).__init__(name, init_config, agentConfig, instances)
+        self.api_versions = {
+            '1': {
+                'functions': {
+                    'check_leader': self.check_leader_v1,
+                    'check_health': self.check_health_v1,
+                }
+            },
+        }
+        self.config = {}
+
+    def check(self, instance):
+        config = self.get_config(instance)
+        if config is None:
+            return
+
+        tags = list(config['tags'])
+
+        try:
+            config['api']['check_leader'](config, tags)
+            config['api']['check_health'](config, tags)
+        except ApiUnreachable:
+            return
+
+        self.service_check(self.SERVICE_CHECK_CONNECT, AgentCheck.OK, tags=tags)
+
+    def check_leader_v1(self, config, tags=None):
```

Comment:
Is it possible for ``tags`` to equal ``None``? Not opposed to leaving it like this.